core/feconf.py

```diff
@@ -40,6 +40,26 @@
 POST_COMMIT_STATUS_PUBLIC = 'public'
 POST_COMMIT_STATUS_PRIVATE = 'private'
 
+CollectionPropertiesType = Tuple[
+    Literal['title'],
+    Literal['category'],
+    Literal['objective'],
+    Literal['language_code'],
+    Literal['tags']
+]
+
+
+class ValidCmdDict(TypedDict):
+    """Dictionary representing valid commands specs."""
+
+    name: str
+    required_attribute_names: List[str]
+    optional_attribute_names: List[str]
+    user_id_attribute_names: List[str]
+    allowed_values: Dict[str, Union[CollectionPropertiesType, List[str]]]
```

Comment:
Union is added here to make it generalised. like for topic_domain.py we just have to add `SUBTOPIC_PROPERTIES`'s tuple and `TOPIC_PROPERTIES`'s tuple in this union and then it will also work for  topic_domain's `ALLOWED_COMMANDS`
Without Union we have to make different validCmdDict. Like CollectionValidCmdDict, TopicValidCmdDict and etc.

Reply:
Are we talking about all the keys ?